Lib/multiprocessing/pool.py

```diff
@@ -220,20 +231,34 @@ def __init__(self, processes=None, initializer=None, initargs=(),
             exitpriority=15
             )
 
+
     @staticmethod
     def _join_exited_workers(pool):
         """Cleanup after any worker processes which have exited due to reaching
         their specified lifetime.  Returns True if any workers were cleaned up.
+        Returns None if the process pool is broken.
         """
         cleaned = False
-        for i in reversed(range(len(pool))):
-            worker = pool[i]
-            if worker.exitcode is not None:
+        broken = []
+        for i, p in reversed(list(enumerate(pool))):
+            broken.append(p.exitcode not in (None, 0))
```

Comment:
You don't need to accumulate values if you're just interested in one of the workers being broken. Instead, you can simply use a boolean variable.